airflow/jobs/scheduler_job_runner.py

```diff
@@ -1348,8 +1354,14 @@ def _update_state(dag: DAG, dag_run: DagRun):
                     tags={"dag_id": dag.dag_id},
                 )
 
+        # cache saves time during scheduling of many dag_runs for same dag
+        cached_dags: dict = {}
+
         for dag_run in dag_runs:
-            dag = dag_run.dag = self.dagbag.get_dag(dag_run.dag_id, session=session)
+            dag = dag_run.dag = get_value_with_cache(
+                cached_dags, dag_run.dag_id, self.dagbag.get_dag, dag_run.dag_id, session=session
+            )
```

Comment:
```suggestion
            dag = dag_run.dag = get_value_with_cache(
                cached_dags, dag_run.dag_id, lambda: self.dagbag.get_dag(dag_run.dag_id, session=session)
            )
```

Reply:
Thanks for the feedback! Using now lambda.